hoomd/update/box_resize.py

```diff
@@ -0,0 +1,104 @@
+from hoomd.operation import _Updater
+from hoomd.box import Box
+from hoomd.parameterdicts import ParameterDict
+from hoomd.typeconverter import OnlyType
+from hoomd.variant import Variant, Power
+from hoomd.util import variant_preprocessing
+from hoomd import _hoomd
+
+
+class BoxResize(_Updater):
+    """Resizes the box between an initial and final box.
+
+    When part of a `Simulation` ``updater`` list, this object will resize the
+    box between the initial and final boxes passed. The behavior is a linear
+    interpolation between the initial and final boxes where the minimum of the
+    variant is the initial box and the maximum is the final box. All
+    values between the minimum and maximum result in a box that is the
+    interpolation of the three lengths and tilt factors of the initial and final
+    boxes.
+
+    Note:
+        The passed `Variant` must be well behaved (i.e. it must have a
+        true minimum and maximum) or the behavior of the updater is undefined.
```

Comment:
Is it even possible to define variants that don't, since those are virtual methods that must be implemented? I'm not quite sure what this means.

Reply:
Yes for instance, one could just do this

````python
class Custom(Variant):
    def _min(self):
        raise RuntimeError("Varinat has no minimum.")
    def _max(self):
        raise RuntimeError("Varinat has no maximum.")
````

Reply:
Ha fair enough. I suppose it's worth trying to protect against that kind of blatantly incorrect behavior, I didn't even consider it.